mongo/session.go

```diff
@@ -37,140 +56,50 @@ var withTransactionTimeout = 120 * time.Second
-	// the TransientTransactionError and UnknownTransactionCommitResult labels are retried for up to
-	// 120 seconds. Inside the callback, the SessionContext must be used as the Context parameter
-	// for any operations that should be part of the transaction. If the ctx parameter already has a
-	// Session attached to it, it will be replaced by this session. The fn callback may be run
-	// multiple times during WithTransaction due to retry attempts, so it must be idempotent.
-	// Non-retryable operation errors or any operation errors that occur after the timeout expires
-	// will be returned without retrying. If the callback fails, the driver will call
-	// AbortTransaction. Because this method must succeed to ensure that server-side resources are
-	// properly cleaned up, context deadlines and cancellations will not be respected during this
-	// call. For a usage example, see the Client.StartSession method documentation.
-	WithTransaction(ctx context.Context, fn func(ctx SessionContext) (interface{}, error),
-		opts ...*options.TransactionOptions) (interface{}, error)
-
-	// EndSession aborts any existing transactions and close the session.
-	EndSession(context.Context)
-
-	// ClusterTime returns the current cluster time document associated with the session.
-	ClusterTime() bson.Raw
-
-	// OperationTime returns the current operation time document associated with the session.
-	OperationTime() *primitive.Timestamp
-
-	// Client the Client associated with the session.
-	Client() *Client
-
-	// ID returns the current ID document associated with the session. The ID document is in the
-	// form {"id": <BSON binary value>}.
-	ID() bson.Raw
-
-	// AdvanceClusterTime advances the cluster time for a session. This method returns an error if
-	// the session has ended.
-	AdvanceClusterTime(bson.Raw) error
-
-	// AdvanceOperationTime advances the operation time for a session. This method returns an error
-	// if the session has ended.
-	AdvanceOperationTime(*primitive.Timestamp) error
-
-	session()
-}
-
-// XSession is an unstable interface for internal use only.
-//
-// Deprecated: This interface is unstable because it provides access to a session.Client object, which exists in the
-// "x" package. It should not be used by applications and may be changed or removed in any release.
-type XSession interface {
-	ClientSession() *session.Client
-}
-
-// sessionImpl represents a set of sequential operations executed by an application that are related in some way.
-type sessionImpl struct {
-	clientSession       *session.Client
-	client              *Client
-	deployment          driver.Deployment
-	didCommitAfterStart bool // true if commit was called after start with no other operations
-}
-
-var _ Session = &sessionImpl{}
-var _ XSession = &sessionImpl{}
-
 // ClientSession implements the XSession interface.
-func (s *sessionImpl) ClientSession() *session.Client {
+func (s *Session) ClientSession() *session.Client {
```

Comment:
Are you suggesting that we remove this method from the stable API? Why would we want to remove access to the client session?

Reply:
`session.Client` is part of the `x/` experimental packages, so it's unusual that it's exposed via this API. Removing it would really be part of the scope of [GODRIVER-2700](https://jira.mongodb.org/browse/GODRIVER-2700), so you can disregard this comment for now.